packages/@webex/internal-plugin-feature/src/feature.js

```diff
@@ -63,11 +64,13 @@ const Feature = WebexCore.WebexPlugin.extend({
    * @returns {undefined}
    */
   listen() {
-    this.listenTo(
-      this.webex.internal.mercury,
-      'event:featureToggle_update',
-      this.handleFeatureUpdate
-    );
+    if (this.mercury) {
+      this.listenTo(this.mercury, 'event:featureToggle_update', this.handleFeatureUpdate);
```

Comment:
this doesn't feel like really fully fixing the circular dependency, as the feature plugin will still be listening on the events from mercury and needs to know what these events are. I think, if we wanted to do this fully we should instead of adding a `setMercury()` method, just add a method that mercury calls when a feature is updated -  it would be basically almost the same as handleFeatureUpdate() method, just the arguments should be changed from `envelope` to `feature` and maybe it could be renamed to `featureUpdated()` or `setFeature()` or `updateFeature()` or something like that

Reply:
hmm.... not sure... but probably should be ok. We don't need that call to `listen()` anymore, because the features plugin will always listen - it will always get called from Mercury with a new feature toggle. The only possible issue is what happens if a feature toggle is updated before web app calls `webex.internal.feature.listen()` - I think with the current code as it is the web app would just miss that event anyway, so with the new code Mercury might call features plugin handleFeatureUpdate() before web app registers the listeners for the feature toggle events, but I think it's just the same is it currently is, and hasn't caused any problems, so probably should be fine.... Can you do some testing with the web app with this change and the SDK changes and change some toggle that web app listens for (for example "web-tcp-reachability") and check that it still works correctly (and configures sdk to do tcp reachability)?

Reply:
I got info from the mercury and the feature Ask spaces and triggered this event handler by updating one of the following features:
* group-message-notifications
* mention-notifications
* thread-notifications

Looks like that event is triggered only for these three legacy toggles. I tested these changes with the WWC changes as well here: https://sqbu-github.cisco.com/WebExSquared/webex-web-client/pull/5975

The event is triggered alright. Kindly re-review

@marcin-bazyl